api/comments.rb

```diff
@@ -1,5 +1,8 @@
 get "#{APIPREFIX}/comments/:comment_id" do |comment_id|
-  comment.to_hash(recursive: bool_recursive).to_json
+  @comment = comment
+  comment_hash = @comment.to_hash(recursive: bool_recursive)
+  verify_or_fix_cached_comment_count(@comment, comment_hash)
```

Comment:
Looks good with your update.  I agree that we don't want to make an extra db call, so we only update when recursive is true, as you have done.